pgrx-sql-entity-graph/src/pg_extern/mod.rs

```diff
@@ -83,12 +83,17 @@ pub struct PgExtern {
 }
 
 impl PgExtern {
+    #[track_caller]
     pub fn new(attr: TokenStream2, item: TokenStream2) -> Result<CodeEnrichment<Self>, syn::Error> {
         let mut attrs = Vec::new();
         let mut to_sql_config: Option<ToSqlConfig> = None;
 
         let parser = Punctuated::<Attribute, Token![,]>::parse_terminated;
-        let punctuated_attrs = parser.parse2(attr)?;
+        let punctuated_attrs = parser.parse2(attr).map_err(|e| {
+            let mut e = e.clone();
+            e.combine(syn::Error::new(Span::call_site(), "lol here we go"));
+            e
+        })?;
```

Comment:
Hey Jubilee, you do this a lot with `syn::Result`s in this PR, have you considered factoring out this error-combining pattern into its own function, since dtolnay is probably not going to make combining one `syn::Error` with another more ergonomic than it already is anytime soon?

Reply:
Done!